File: leetcode/weekly/352/c/c.cpp
class Solution {
public:
    long long continuousSubarrays(vector<int> &nums) {
        long long ans = 0;
        multiset<int> s;
        int left = 0, n = nums.size();
        for (int right = 0; right < n; right++) {
            s.insert(nums[right]);
            while (*s.rbegin() - *s.begin() > 2)
                s.erase(s.find(nums[left++]));
                // s.erase(nums[left])  这样写会删除所有nums[left]
            ans += right - left + 1;
        }
        return ans;
    }
};

File: leetcode/weekly/352/c/c.py
# Ë«Ö¸Õë + Set / Counter 
class Solution:
    def continuousSubarrays(self, nums: List[int]) -> int:
        ans = left = 0
        cnt = Counter()
        for right,x in enumerate(nums):
            cnt[x]+=1
            while max(cnt)-min(cnt) > 2:
                y = nums[left]
                cnt[y]-=1
                if cnt[y] == 0:
                    del cnt[y]
                left+=1
            ans += right-left+1
        return ans